src/others/file_explorer/lv_file_explorer.h

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file lv_file_explorer.h
+ *
+ */
+
+#ifndef LV_FILE_EXPLORER_H
+#define LV_FILE_EXPLORER_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+
+#if LV_USE_FILE_EXPLORER != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef enum {
+    LV_EXPLORER_SORT_NONE,
+    LV_EXPLORER_SORT_KIND,
+} lv_file_explorer_sort_t;
+
+#if LV_FILE_EXPLORER_QUICK_ACCESS
+typedef enum {
+    LV_EXPLORER_HOME_DIR,
+    LV_EXPLORER_MUSIC_DIR,
+    LV_EXPLORER_PICTURES_DIR,
+    LV_EXPLORER_VIDEO_DIR,
+    LV_EXPLORER_DOCS_DIR,
+    LV_EXPLORER_FS_DIR,
+} lv_file_explorer_dir_t;
+#endif
+
+/*Data of canvas*/
+typedef struct {
+    lv_obj_t obj;
+    lv_obj_t * cont;
+    lv_obj_t * head_area;
+    lv_obj_t * browser_area;
+    lv_obj_t * file_table;
+    lv_obj_t * path_label;
+#if LV_FILE_EXPLORER_QUICK_ACCESS
+    lv_obj_t * quick_access_area;
+    lv_obj_t * list_device;
+    lv_obj_t * list_places;
+    char * home_dir;
+    char * music_dir;
+    char * pictures_dir;
+    char * video_dir;
+    char * docs_dir;
+    char * fs_dir;
+#endif
+    const char * sel_fn;
+    char   current_path[LV_FILE_EXPLORER_PATH_MAX_LEN];
+    lv_file_explorer_sort_t sort;
+} lv_file_explorer_t;
```

Comment:
Why is the `lv_file_explorer_t` struct **definition** on the H file?
I would suggest putting it in the C file, only keeping the **declaration** on the H file if needed at all (`typedef struct lv_file_explorer_t;`)
This is also important for keeping the Micropython API smaller because there is no need to provide API and accessor logic for each field in this struct.

Reply:
We expose these "classes" because it might be used by derived classes. So I vote for keeping it public. 

It's a long lasting discussion with @amirgon about how to keep the API smaller. If we want global change, I'm coming around to the idea of having e.g. `lv_btn.h` and `lv_btn_priv.h`. In C `*_priv.h` can be included in C files where needed.

Reply:
> > I was trying to fix some of them in #3635 but I think it's better to make sure all tests are passing and CI is clean before merging in the first place.
> 
> My bad, okay.
> 
> So for this PR both failing CI is unrelated, right? So can it be merged?

The two problems of CI are not involved in my PR. I think it should be fixed in other PR.

Reply:
@YobeZhou I believe that if you rebase this PR on latest LVGL these problems will be gone.

Reply:
> @YobeZhou I believe that if you rebase this PR on latest LVGL these problems will be gone.

If necessary, I will consider resubmitting PR after all discussions are completed. Before that, I wanted to know if I needed to do this? @kisvegabor

Reply:
Let leave the `lv_file_explorer_t` in the header file. If we change it we should change other widgets too.